### ISSUE TITLE
The date range filter is not working

### ISSUE THREAD

**Issue by pyroAdrian** [10up/ElasticPress#259]
Hi,

I'm trying to filter the posts using date range and it seems that is not working. Using this query arguments:
```
    $args = array(
        's' => 'festivals',
        'date_query' => array(
            array(
                'after' => 'April 1st, 2015',
                'before' => array(
                    'year' => 2015,
                    'month' =>5,
                    'day' => 1
                   ),
                'inclusive' => false,
            ),
        )
    );
```
the code is generating the following json that is sent to elasticsearch
```
{
    "from": 0,
    "size": 16,
    "sort": [{
        "_score": {
            "order": "desc"
        }
    }],
    "query": {
        "bool": {
            "should": [{
                "multi_match": {
                    "query": "festival",
                    "fields": ["post_title", "post_excerpt", "post_content"],
                    "boost": 2
                }
            }, {
                "fuzzy_like_this": {
                    "fields": ["post_title", "post_excerpt", "post_content"],
                    "like_text": "festival",
                    "min_similarity": 0.75
                }
            }]
        }
    },
    "filter": {
        "and": [{
            "bool": {
                "must": [{
                    "range": {
                        "post_date": {
                            "gte": "2015-04-01 00:00:00",
                            "lte": "2015-05-01 00:00:00"
                        }
                    }
                }]
            }
        }, {
            "term": {
                "post_type": ["post"]
            }
        }]
    }
}
```

Does anyone know what could be the issue ?

Thank you

**Comment by EduardMaghakyan** [comment]
@pyroAdrian just to confirm that we are on the same page you mean that this part

```php
"post_date": {
                            "gte": "2015-04-01 00:00:00",
                            "lte": "2015-05-01 00:00:00"
                        }
```

should be

```php
"post_date": {
"gt": "2015-04-01 00:00:00",
"lt": "2015-05-01 00:00:00"
}
```

because inclusive is set to false?

**Comment by pyroAdrian** [comment]
Hi @EduardMaghakyan ,

I've found what was the main cause of the issue. The problem was that I've indexed the data using "wp elasticpress index" and this command is indexing the post_date as string and the filter is not working correctly if the field is string.

I've indexed the data using the mapping (index --setup) and the post_date has the type "date" and filter worked.

This issue can be closed.
